runtime/src/launch/aprun/aprun-utils.c

```diff
@@ -201,6 +202,22 @@ int getCPUsPerCU() {
   return numCPUsPerCU;
 }
 
+const char *getNodeListStr() {
+  return getAprunArgStr(aprun_L);
+}
+char *getNodeListOpt() {
+  const char *nodeList = getenv("CHPL_LAUNCHER_NODELIST");
+  char *nodeListOpt = NULL;
+
+  if (nodeList) {
+    nodeListOpt = sys_malloc(strlen(getNodeListStr()) + strlen(nodeList) + 1);
```

Comment:
You'll actually want to use `chpl_mem_allocMany` instead of `sys_malloc`. chpl_mem_allocMany will map to sys_malloc for launchers, but it's the "user" facing API. Alternatively, I think similar cases in other launchers just use a fixed buffer and truncate or error if the string is too long